python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -3259,103 +2932,17 @@ class DPExplainableBoostingRegressor(EBMModel, RegressorMixin, ExplainerMixin):
     noise_scale_boosting\\_ : float
         The noise scale during boosting.
     """
+    objective: Literal["rmse"] = "rmse"
 
-    n_features_in_: int
-    term_names_: List[str]
-    bins_: List[Union[List[Dict[str, int]], List[np.ndarray]]]  # np.float64, 1D[cut]
-    feature_names_in_: List[str]
-    feature_types_in_: List[str]
-    feature_bounds_: np.ndarray  # np.float64, 2D[feature, min_max]
-    term_features_: List[Tuple[int, ...]]
-    bin_weights_: List[np.ndarray]  # np.float64, [bin0...]
-    bagged_scores_: List[np.ndarray]  # np.float64, [bag, bin0..., ?class]
-    term_scores_: List[np.ndarray]  # np.float64, [bin0..., ?class]
-    standard_deviations_: List[np.ndarray]  # np.float64, [bin0..., ?class]
-    link_: str
-    link_param_: float
-    bag_weights_: np.ndarray  # np.float64, 1D[bag]
-    breakpoint_iteration_: np.ndarray  # np.int64, 2D[stage, bag]
-
-    noise_scale_binning_: float
-    noise_scale_boosting_: float
-
-    intercept_: float
-    bagged_intercept_: np.ndarray  # np.float64, 1D[bag]
-    min_target_: float
-    max_target_: float
+    min_target_: float = field(init=False)
+    max_target_: float = field(init=False)
 
     # TODO PK v.3 use underscores here like RegressorMixin._estimator_type?
-    available_explanations = ["global", "local"]
-    explainer_type = "model"
+    available_explanations: ClassVar[List[str]] = ["global", "local"]
+    explainer_type: ClassVar[str] = "model"
 
     """ Public facing DPEBM regressor."""
 
-    def __init__(
-        self,
-        # Explainer
-        feature_names: Optional[Sequence[Union[None, str]]] = None,
-        feature_types: Optional[
-            Sequence[Union[None, str, Sequence[str], Sequence[float]]]
-        ] = None,
-        # Preprocessor
-        max_bins: int = 32,
-        # Stages
-        exclude: Optional[Sequence[Union[int, str, Sequence[Union[int, str]]]]] = None,
-        # Ensemble
-        validation_size: Optional[Union[int, float]] = 0,
-        outer_bags: int = 1,
-        # Boosting
-        learning_rate: float = 0.01,
-        max_rounds: Optional[int] = 300,
-        # Trees
-        max_leaves: int = 3,
-        # Overall
-        n_jobs: Optional[int] = -2,
-        random_state: Optional[int] = None,
-        # Differential Privacy
-        epsilon: float = 1.0,
-        delta: float = 1e-5,
-        composition: str = "gdp",
-        bin_budget_frac: float = 0.1,
-        privacy_bounds: Optional[
-            Union[np.ndarray, Mapping[Union[int, str], Tuple[float, float]]]
-        ] = None,
-        privacy_target_min: Optional[float] = None,
```

Comment:
Is there any way to preserve the parameter ordering of the \_\_init\_\_ functions if some of the attributes are defined in the EBMBase, EBMModel, and ExplainableBoostingClassifier/ExplainableBoostingRegressor classes while still using the dataclass decorators?  It looks by default like the parameters will be grouped by inheritance ordering, so all the EBMBase attributes would come first, then all the EBMModel attributes, etc..